processors/HNL_new.py

```diff
@@ -0,0 +1,559 @@
+                globalFeatures=[],
+                outputName="selectedL1PreFiringJets_"+systName,
+                globalOptions=globalOptions
+            )
+        ])
+    
+        if isMC:
+            sequence.append(
+                JetTruthFlags(
+                    inputCollection=lambda event, systName=systName: getattr(event, "selectedJets_"+systName),
+                    originVariables = ['displacement_xy'],
+                    outputName="selectedJets_"+systName,
+                    globalOptions=globalOptions
+                )
+            )
+    
+    systNames = jetDict.keys()
+    sequence.append(
+        EventSkim(selection=lambda event, systNames=systNames: 
+            any([getattr(event, "nselectedJets_"+systName) > 0 for systName in systNames])
+        )
+    )
+            
+    return sequence
+    
+    
+def eventReconstructionSequence(jetMetDict):
+    sequence = []
+    for systName,(jetCollection,metObject) in jetMetDict.items():
+        sequence.extend([
+            EventObservables(
+                lepton1Object=lambda event: event.leadingLeptons[0],
+                lepton2Object=None if args.leptons==1 else (lambda event: event.subleadingLeptons[0]),
+                jetCollection=jetCollection,
+                metInput=metObject,
+                globalOptions=globalOptions,
+                outputName=systName
+            ),
+
+            HNLReconstruction(
+                lepton1Object=lambda event: event.leadingLeptons[0],
+                lepton2Object=None if args.leptons==1 else (lambda event: event.subleadingLeptons[0]),
+                jetCollection=jetCollection,
+                globalOptions=globalOptions,
+                outputName=systName,
+            )
+        ])
+        
+    return sequence
+    
+    
+def taggerSequence(jetDict, modelFile, taggerName):
+    sequence = []
+    sequence.append(
+        TaggerEvaluationProfiled(
+            modelPath=modelFile,
+            featureDictFile=featureDictFile,
+            inputCollections=jetDict.values(),
+            taggerName=taggerName,
+            profiledLabelDict = {
+                'LLP_Q': ['LLP_Q','LLP_QTAU_H','LLP_QTAU_3H'],
+                'LLP_QE': [ 'LLP_QE'],
+                'LLP_QMU': [ 'LLP_QMU'],
+                'LLP_QTAU_3H': ['LLP_QTAU_3H']
+            },
+            globalOptions=globalOptions,
+            evalValues = np.linspace(-1.9,1.9,5*4),
+        )
+    )
+   
+    for systName,jetCollection in jetDict.items():
+        sequence.append(
+            JetTaggerProfiledResult(
+                inputCollection=jetCollection,
+                taggerName=taggerName,
+                outputName="hnlJet_"+systName,
+                profiledLabels = ['LLP_Q','LLP_QE','LLP_QMU','LLP_QTAU_3H'],
+                kinds = ['ratio'],
+                globalOptions={"isData": False}
+            )
+        )
+    return sequence
+
+    
+
+def bdtSequence(systematics):
```

Comment:
We want to keep correlated and uncorrelated BDT models for now

Reply:
not sure if the correlated BDT is a good idea; but can be easily added with this new code